### PR TITLE
make places of DataLoader optional.

### PR DESCRIPTION
**Pull request by heavengate**
### PR types
Function optimization

### PR changes
APIs

### Describe
make `places` argument in `paddle.io.DataLoader` optional
- if `place` not set(None), use `_current_expected_place()` instead

dataloder

屏幕快照 2020-09-24 下午5 01 54
屏幕快照 2020-09-24 下午5 02 07
屏幕快照 2020-09-24 下午5 02 18

### CODE REVIEW

File: python/paddle/fluid/reader.py [hunk 1]
```diff
@@ -167,10 +167,10 @@ class DataLoader(object):
             The variables should be created by :code:`fluid.data()`.
             :attr:`feed_list` must be set if :attr:`return_list` is
             False. Default None.
-        places(list(Place)|tuple(Place)): a list of Place, to put data
-            onto, :attr:`places` must be set in both static graph and 
-            dynamic graph mode, in dynamic graph mode, place number must
-            be 1. Default None.
+        places(list(Place)|tuple(Place)|optional): a list of Place,
+            to put data onto, :attr:`places` should be None in dynamic
+            graph mode. if :attr:`places` is None in static mode, default
```
Comment (chenwhql) [inline python/paddle/fluid/reader.py]: should be None? maybe `can be None` is better?

Comment (chenwhql) [inline python/paddle/fluid/reader.py]: `in static mode`?  not dynamic?

Comment (heavengate) [reply python/paddle/fluid/reader.py]: Done, thanks!

Comment (heavengate) [reply python/paddle/fluid/reader.py]: Done, thanks!

**Review by chenwhql — DISMISSED** [review]

**Review by saxon-zh — DISMISSED** [review]
LGTM

**Review by chenwhql — APPROVED** [review]
LGTM

**Review by saxon-zh — APPROVED** [review]
LGTM

**Pull request merged**